src/cmake/compiler.cmake

```diff
@@ -192,6 +193,13 @@ if (USE_LIBCPLUSPLUS AND CMAKE_COMPILER_IS_CLANG)
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
 endif ()
 
+# GCC 5+: honor build-time option for whether or not to use new string ABI
+if (CMAKE_COMPILER_IS_GNUCC AND NOT ${GCC_VERSION} VERSION_LESS 5.0)
+    if (NOT ${GLIBCXX_USE_CXX11_ABI} STREQUAL "")
+        add_definitions ("-D_GLIBCXX_USE_CXX11_ABI=${GLIBCXX_USE_CXX11_ABI}")
```

Comment:
I don't think this is related to which compiler you use, just if you are using `libstdc++` or not.

It applies just as well to clang or others.

Reply:
Yes, that's how I originally had it! And it caused endless heartache and build breaks when using clang. The only way I could find to keep this a small task was to restrict it to gcc and resolve to return to the clang issue if and when it popped up as a real problem.

Reply:
What kind of errors where you getting?

Maybe leave a comment to that effect at least so we remember why its scoped with the compiler?

Reply:
I was getting errors like headers were strangely mismatched.

I added a big FIXME comment here, so we'll know to come back to this spot if a case arises where we need to fix this for clang.